cmd/stack_lint.go

```
@@ -0,0 +1,135 @@
+
+		if fileDoesNotExist(stackPath+"/README.md") != nil {
+			Info.log("ERROR: Missing README.md in: " + stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(stackPath+"/stack.yaml") != nil {
+			Info.log("ERROR: Missing stack.yaml in: " + stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath) != nil {
+			Info.log("ERROR: Missing image directory in " + stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath+"/Dockerfile-stack") != nil {
+			Info.log("ERROR: Missing Dockerfile-stack in " + imagePath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath+"/LICENSE") != nil {
+			Info.log("ERROR: Missing LICENSE in " + imagePath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(configPath) != nil {
+			Info.log("WARNING: Missing config directory in " + imagePath + " (Knative deployment will be used over Kubernetes)")
+			warningCount++
+
+		}
+
+		if fileDoesNotExist(configPath+"/app-deploy.yaml") != nil {
+			Info.log("WARNING: Missing app-deploy.yaml in " + configPath + " (Knative deployment will be used over Kubernetes)")
+			warningCount++
+		}
+
+		if fileDoesNotExist(projectPath+"/Dockerfile") != nil {
+			Info.log("WARNING: Missing Dockerfile in " + projectPath)
+			warningCount++
+		}
+
+		if fileDoesNotExist(templatePath) != nil {
+			Info.log("ERROR: Missing template directory in: " + stackPath)
+			errorCount++
+		}
+
+		if IsEmptyDir(templatePath) != nil {
+			Info.log("ERROR: No templates found in: " + templatePath)
+			errorCount++
+		}
+
+		templates, _ := ioutil.ReadDir(templatePath)
+		for _, f := range templates {
+			if fileDoesNotExist(templatePath+"/"+f.Name()+"/"+".appsody-config.yaml") == nil {
+				Info.log("ERROR: Unexpected .appsody-config.yaml in " + templatePath + "/" + f.Name())
+				errorCount++
+			}
+		}
+
+		if errorCount > 0 {
+			Info.log("\nLINT TEST FAILED")
+			Info.log("\nTOTAL ERRORS: " + strconv.Itoa(errorCount))
+			Info.log("TOTAL WARNINGS: " + strconv.Itoa(warningCount))
+
```

If there are errors, you should return an error object, instead of nil. This ensures that the command returns a non-zero error code so other tools like Travis or test scripts know to fail